pkg/kds/global/components.go

```diff
@@ -114,7 +114,7 @@ func Setup(rt runtime.Runtime) error {
 
 	onGlobalToZoneSyncConnect := mux.OnGlobalToZoneSyncConnectFunc(func(stream mesh_proto.KDSSyncService_GlobalToZoneSyncServer, errChan chan error) {
 		zoneID, err := util.ClientIDFromIncomingCtx(stream.Context())
-		if err != nil {
+		if err != nil && !errors.Is(err, context.Canceled) {
```

Comment:
I think it's not fully correct. Here for example we won't return on context cancelled error from `util.ClientIDFromIncomingCtx`

Reply:
you are right, removed. As for the rest changes in this review, there could be context cancelled error returned